Aztec/Classes/Extensions/NSAttributedString+Analyzers.swift

```diff
@@ -6,6 +6,22 @@ import UIKit
 //
 extension NSAttributedString {
 
+    /// Returns true if the text preceding a given location contains the specified attribute key.
+    ///
+    /// - Parameters:
+    ///   - location: location to check
+    ///   - key: the attributed to check
+    /// - Returns: True if the attribute is there
+    ///
+    func isLocation(_ location: Int, preceededBy key: NSAttributedStringKey) -> Bool {
+        let beforeRange = NSRange(location: location - 1, length: 1)
+        guard beforeRange.location >= 0 else {
```

Comment:
We could guard against `location == 0` on top of the method, since there's no need to create the range in this scenario.